crates/chain/tests/test_keychain_txout_index.rs

```diff
@@ -210,13 +210,44 @@ fn test_lookahead() {
             last_internal_index as usize + 1,
         );
     }
+}
 
-    // when:
-    // - scanning txouts with spks above last stored index
-    // expect:
-    // - cached scripts count should increase as expected
-    // - last stored index should increase as expected
-    // TODO!
+// when:
+// - scanning txouts with spks above last stored index
+// expect:
+// - last revealed index should increase as expected
+// - last used index should change as expected
+#[test]
+fn test_scan_with_lookahead() {
+    let (mut txout_index, external_desc, _) = init_txout_index();
+    txout_index.set_lookahead_for_all(10);
+
+    let spks: BTreeMap<u32, Script> = [0, 10, 20, 30]
```

Comment:
:bulb: after the loop try with 41 and check that it doesn't find it

Reply:
Good idea